ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PlanNode.java

```diff
@@ -192,10 +195,13 @@ static void throwKeysNotIncludedError(
       final LogicalSchema schema
   ) {
     // When doing a `select *` key columns should be at the front of the column list
-    // but are added at the back during processing for performance reasons.
-    // Switch them around here:
-    final Stream<Column> keys = columns.stream()
-        .filter(c -> schema.isKeyColumn(c.name()));
+    // but are added at the back during processing for performance reasons. Furthermore,
+    // they should be selected in the order of the key, if that is not the same as the
+    // ordering within the value. Switch them around here:
+    final ImmutableMap<ColumnName, Column> columnsByName = Maps.uniqueIndex(columns, Column::name);
+    final Stream<Column> keys = schema.key().stream()
+        .map(key -> columnsByName.get(key.name()))
+        .filter(Objects::nonNull);
```

Comment:
Wondering why we need this filter?

Reply:
um, we might not need this one here after digging into it more - but from the API of this method, there's nothing prevent `columns` from not covering everything in `schema`, which would mean `columnsByName.get(key.name())` would return empty. Without refactoring this method signature I think it's better to be defensive here.

Reply:
Reading further below, I feel this logic need to be kept as-is even when we have https://github.com/confluentinc/ksql/issues/6374 in place with a clear separation (I'm assuming `schema` references the physical schema of the source, and `columns` references the logical schema mapped from the phyiscal schema by then).

Reply:
> I think it's better to be defensive here.

I guess that is the question? We know that it should never be `null`, so should not fail fast to expose a bug?